README.md

```diff
@@ -121,54 +98,72 @@ partial interface WorkerNavigator {
 ```WebIDL
 [Exposed=(DedicatedWorker,SharedWorker,Window), SecureContext]
 interface SmartCardResourceManager {
-  // Returns all currently available readers.
-  Promise<sequence<SmartCardReader>> getReaders();
-
-  // Rejects with "NotSupportedError" if the feature is not supported
-  // by the Operating System's PC/SC stack.
-  Promise<SmartCardReaderPresenceObserver> watchForReaders();
+  // Requests a PC/SC context from the platform's PC/SC stack.
+  Promise<SmartCardContext> establishContext();
 };
 ```
-### SmartCardReaderPresenceObserver
-```WebIDL
-interface SmartCardReaderPresenceObserver : EventTarget {
-  // Emits SmartCardReaderPresenceEvent.
-  attribute EventHandler onreaderadd;
-  attribute EventHandler onreaderremove;
-
-  // Emits SmartCardErrorEvent.
-  // Occurs when an error prevents the monitoring of readers being added or
-  // removed. Once an observer receives an error event it will no longer
-  // receive any other events.
-  // If an application wants to retry, it can call smartCard.watchForReaders()
-  // again to acquire a new observer.
-  attribute EventHandler onerror;
-};
-```
-### SmartCardReader
+### SmartCardContext
+
 ```WebIDL
-interface SmartCardReader : EventTarget {
-  readonly attribute DOMString name;
+// A context for communicating with the PC/SC resource manager.
+interface SmartCardContext {
+  Promise<sequence<DOMString>> listReaders();
 
-  // Connects to the card in the reader.
-  // The user agent may display a permission prompt.
-  Promise<SmartCardConnection> connect(SmartCardAccessMode accessMode,
+  Promise<sequence<SmartCardReaderStateOut>> getStatusChange(
+      sequence<SmartCardReaderStateIn> readerStates,
+      optional SmartCardGetStatusChangeOptions options = {});
+
+  Promise<SmartCardConnectResult> connect(
+      DOMString readerName,
+      SmartCardAccessMode accessMode,
       optional sequence<SmartCardProtocol> preferredProtocols);
+};
+
+dictionary SmartCardReaderStateIn {
+  required DOMString readerName;
+  required SmartCardReaderStateFlags currentState;
+};
+
+dictionary SmartCardReaderStateOut {
+  required DOMString readerName;
+  required SmartCardReaderStateFlags eventState;
+  required ArrayBuffer answerToReset;
+};
+
+dictionary SmartCardGetStatusChangeOptions {
+  // Timeout given to the platform's PC/SC GetStatusChange call.
+  DOMHighResTimeStamp timeout;
```

Comment:
Do we need to provide a separate timeout functionality given developers can pass `AbortSignal.timeout()`?

Reply:
It seems like `AbortSignal.any()` is close enough to shipping we can simply depend on it, and polyfilling the combination of an `AbortController` with an automatic timeout is also trivial.